### DISCUSSION TITLE
Display progress bar when installing an update.

### DISCUSSION THREAD

**Discussion by joaochris** [sircharlo/meeting-media-manager#525] [Ideas] [answered]
It would be nice, if when updating the program, it showed a progress bar, and a button for us to run the application if we want.

I currently see that the only way we know the update has been installed is when the logo changes from a blank sheet of paper to the application.

Would it be possible to do this?

**Comment by mtdvlpr** [comment, marked as answer]
I will look into it! It will depend on whether or not [electron-builder](https://www.electron.build/) supports it

**Reply by mtdvlpr** [reply]
I can't find anything about it. But what I added now is a "install now" option on the update downloaded notification. When you press that it should install it with a install progress window. I can't test it however, so we'll find out how well it works in the release after the next release :)

**Reply by mtdvlpr** [reply]
@sircharlo, do you know of a way to test the auto-updater before releasing a new version?

**Reply by sircharlo** [reply]
No, that was always one of my great annoyances. The auto-updater is very hard to test in non-production environments

**Reply by mtdvlpr** [reply]
@sircharlo, what I now did was change the package version to an old one. Build the app locally, install it and then it will say an update is available and you can test the new functionality :)

**Reply by mtdvlpr** [reply]
@joaochris, it works btw! When you press the install now button you will get an update window with progress and once it is complete it will start the application again.

**Reply by joaochris** [reply]
> @joaochris, it works btw! When you press the install now button you will get an update window with progress and once it is complete it will start the application again.

Great! Thank you very much! ✌️

**Reply by joaochris** [reply]
@mtdvlpr I couldn't see the progress bar 🤔

**Reply by sircharlo** [reply]
Did the autoupdate work correctly though?

**Reply by mtdvlpr** [reply]
> @mtdvlpr I couldn't see the progress bar thinking

You are correct. This feature has now been implemented, but you won't see it until the next release. So for the next release, under the notification that a new version has been downloaded and ready to be installed, you will see a blue "install now" button. When you use that button to install the update (instead of restarting the app yourself), you will see a progress bar.

**Reply by joaochris** [reply]
> Did the autoupdate work correctly though?

@sircharlo  Yep!
